### ISSUE TITLE
Antivirus slows down rapidly upload

### ISSUE THREAD

**Issue by Mysterycz** [owncloud/files_antivirus#152]
Hello guys,

I noticed when I turn antivirus on, it slows down all uploads. For example when I upload > 1GB file via web interface, app is showing 0 sec and waiting there for another 3-5 mins ... Worse scenario is upload via mobile app ->

I was uploading some big files (1 GB) WITHOUT antivirus on 100/100 Mbit and it takes like 1-2 minutes. When I turn antivirus ON, it takes 40 minutes to upload only 20% of this file, yes 40 min. Do you know where could be the problem? I am running my cloud server in VM, server has good hardware resources (Like many cores, enough RAM and RAID on server HDDs (not SSDs). 

Thank you :-)

**Comment by VicDeo** [comment]
@Mysterycz 
Most likely you hit https://github.com/owncloud/files_antivirus/issues/125
Check if your owncloud.log has a huge number of entries produced by the files_antivirus app

**Comment by scorpaanon** [comment]
Thats also because the antivirus is scanning the file as you upload its one of the ways that the program works :)

**Comment by Mysterycz** [comment]
@VicDeo I can see many messages in log file like:

... "files_antivirus","message":"OCA\\Files_Antivirus\\BackgroundScanner::run, exception: Can not decrypt this file, probably this is a shared file. Please ask the file owner to reshare the file with you.","level":3,"time":"2017-02-28T10:21:51+00:00","method":"GET","url":"\/cron.php" ...
... "app":"no app in context","message":"Exception: {\"Exception\":\"OC\\\\Encryption\\\\Exceptions\\\\DecryptionFailedException\",\"Message\":\"Encryption library: Decryption (symmetric) of content failed: error:0607A082:digital envelope routines:EVP_CIPHER_CTX_set_key_length:invalid key length\",\"Code\ ...

Nothing more ...

@scorpaanon That's because the antivirus is scanning the file and it slows down upload from 2 minutes to 2 hours? Do you think this is correct behaviour?

**Comment by VicDeo** [comment]
@Mysterycz these messages are not related to upload. They come from background job and encryption app.
Check how many entries like this one in your log:
```
{"reqId":"QpaPE1IicR\/hpupFjzYf","remoteAddr":"","app":"PHP","message":"fwrite(): send of 4 bytes failed with errno=32 Broken pipe at \/var\/www\/owncloud\/apps3\/files_antivirus\/lib\/scanner.php#120","level":3,"time":"2016-11-05T15:25:56+00:00","method":"--","url":"--","user":"admin"}
```

**Comment by Mysterycz** [comment]
@VicDeo I double checked my log and there is not any entry like this one. Only encryption errors.

**Comment by scorpaanon** [comment]
Your file probably has a incorrect hash value. Thats whats giving you the error

**Comment by Mysterycz** [comment]
Can you try to turn antivirus on and upload some big file (1GB+) via Android app to see how long does it take? Then turn antivirus off and check upload time again? To see if this slow upload is only on my OC instance.

**Comment by scorpaanon** [comment]
@Mysterycz  2hours on my phone

**Comment by scorpaanon** [comment]
and thats if i accidentally hit #125

**Comment by VicDeo** [comment]
@Mysterycz it was probably related to scanning individual chunks for large files.
Newly released v0.12.0  skips chunks and scans the complete file only. So there should be a speed improvement.
In addition it has an option to limit filesize for scanning.
